sql/planbuilder/dml.go

```diff
@@ -220,23 +220,29 @@ func (b *Builder) assignmentExprsToExpressions(inScope *scope, e ast.AssignmentE
 		startWinCnt = len(inScope.windowFuncs)
 	}
 
-	tableSch := inScope.node.Schema()
+	tableSch := b.resolveSchemaDefaults(inScope, inScope.node.Schema())
 
 	for i, updateExpr := range e {
 		colName := b.buildScalar(inScope, updateExpr.Name)
 
-		// Prevent update of generated columns
+		innerExpr := b.buildScalar(inScope, updateExpr.Expr)
 		if gf, ok := colName.(*expression.GetField); ok {
 			colIdx := tableSch.IndexOfColName(gf.Name())
 			// TODO: during trigger parsing the table in the node is unresolved, so we need this additional bounds check
 			//  This means that trigger execution will be able to update generated columns
+			// Prevent update of generated columns
 			if colIdx >= 0 && tableSch[colIdx].Generated != nil {
 				err := sql.ErrGeneratedColumnValue.New(tableSch[colIdx].Name, inScope.node.(sql.NameableNode).Name())
 				b.handleErr(err)
 			}
+			// Replace update expressions with default
```

Comment:
the ordering/nesting here seems weird, can we resolve the default during buildScalar? or separately from this if block? or does it really depend on colName?

Reply:
I just reused logic from InsertDestination. We seem to treat `ast.Default` as a placeholder to be replace with whatever is stored in the schema.

BuildScalar would need the resolved schema

I could separate this into its own block, but it would look the same as the block its already in